lib/web_ui/lib/src/engine/view_embedder/display_dpr_stream.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+
+import 'package:ui/src/engine/display.dart';
+import 'package:ui/src/engine/dom.dart';
+import 'package:ui/ui.dart' as ui show Display;
+
+/// Determines if high contrast is enabled using media query 'forced-colors: active' for Windows
+class DisplayDprStream {
+  DisplayDprStream(ui.Display display) : _currentDpr = display.devicePixelRatio {
+    // Start listening to DPR changes.
+    _subscribeToMediaQuery();
+  }
+
+  /// A singleton instance of DisplayDprObserver.
+  static DisplayDprStream instance = DisplayDprStream(EngineFlutterDisplay.instance);
+
+  // Last reported value of DPR.
+  double _currentDpr;
+
+  // Controls the [dprChanged] Stream.
+  final StreamController<double> _dprStreamController = StreamController<double>.broadcast();
+
+  // Listens for the `_currentDpr` to change.
+  late DomMediaQueryList _dprMediaQuery;
+
+  // Creates the media query for the latest known DPR value, and adds a change listener to it.
+  void _subscribeToMediaQuery() {
+    _dprMediaQuery = domWindow.matchMedia('(resolution: ${_currentDpr}dppx)');
+    _dprMediaQuery.addEventListenerWithOptions(
+      'change',
+      createDomEventListener(_onDprMediaQueryChange),
+      <String, Object>{
+        'once': true,
+        'passive': true,
+      });
+  }
+
+  // Handler of the 'change' event.
+  //
+  // This calls subscribe again because events are listened to with `once: true`.
```

Comment:
It would be nice to explain why we are using `once: true`, I'm curious :)

Reply:
I was reading [this example](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio#javascript_2) and I thought that with `once` I wouldn't have to `removeEventListener` all the time?

Reply:
🤦‍♂️ nvm. I just realized that you have to create a new media query for each new value of dpr.